### ISSUE TITLE
`factor_absolute` sometimes fails to fully factorize

### ISSUE THREAD

**Issue by fingolfin** [oscar-system/Oscar.jl#2081]
Noticed by @benlorenz when the CI for PR #2077 [failed](https://github.com/oscar-system/Oscar.jl/actions/runs/4448347367/jobs/7811050994?pr=2077). But I can also reproduce this locally, by re-invoking it a couple of times:
```
julia> Qa, a = polynomial_ring(QQ, :a=>1:2);

julia> R, X = polynomial_ring(fraction_field(Qa), :X=>1:2);

julia> f = factor_absolute((X[1]^2+a[1]*X[2]^2)*(X[1]+2*X[2]+3*a[1]+4*a[2]))
3-element Vector{Any}:
 1
 (X[1] + 2*X[2] + 3*a[1] + 4*a[2], 1)
 (X[1] + t*X[2], X[1] - t*X[2], 1)

[... repeat last command ...]

julia> f = factor_absolute((X[1]^2+a[1]*X[2]^2)*(X[1]+2*X[2]+3*a[1]+4*a[2]))
3-element Vector{Any}:
 1
 (X[1] + 2*X[2] + 3*a[1] + 4*a[2], 1)
 (X[1] + t*X[2], X[1] - t*X[2], 1)

julia> f = factor_absolute((X[1]^2+a[1]*X[2]^2)*(X[1]+2*X[2]+3*a[1]+4*a[2]))
3-element Vector{Any}:
 1
 (X[1] + 2*X[2] + 3*a[1] + 4*a[2], 1)
 (X[1]^2 + a[1]*X[2]^2, 1)
```
According to the docstring, this last return value indicates that `X[1]^2 + a[1]*X[2]^2` is absolutely irreducible. But it isn't.

Oh, and the docstring is slightly incomplete, too: it claims the return value is a vector with one coefficient, and then tuples containing one or two polynomials. Which is technically correct, but it omits the information that those tuples have one extra member, which I think is a multiplicity, maybe?

**Comment by fieker** [comment]
The current doc string (for this signature) mentiones the signature - but the sentence is hard to parse...

**Comment by fingolfin** [comment]
This has been fixed
